### PR TITLE
Tracing fix

### PR DESCRIPTION
**Pull request by JonathanWoollett-Light**
## Changes

Improves tracing documentation and fixes dependency issue.

## Reason

When running `cargo build --features tracing` with `src/utils`
instrumented errors are produced due to the optional dependency not
being enabled. This makes the dependency non-optional to fix this, when
unneeded the dependency will still be ommited at compile time however
this may very slightly increase compile time.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by ShadowCurse — DISMISSED** [review]

File: src/utils/Cargo.toml [hunk 1]
```diff
@@ -18,10 +18,10 @@ versionize = "0.1.10"
 versionize_derive = "0.1.6"
 vmm-sys-util = "0.11.2"
 vm-memory = { version = "0.13.0", features = ["backend-mmap", "backend-bitmap"] }
-log-instrument = { path = "../log-instrument", optional = true }
+log-instrument = { path = "../log-instrument" }
```
Comment (kalyazin) [inline src/utils/Cargo.toml:21]: can we have a regression test that verifies that?

File: docs/tracing.md [hunk 2]
```diff
@@ -33,16 +33,35 @@ outputting a `Trace` level log when entering and exiting every function.
...
+You can run `clippy-tracing --help` for help.
+
 To enable tracing in Firecracker, add instrumentation with:
 
 ```
 clippy-tracing \
 --action fix \
 --path ./src \
---exclude vmm_config/logger.rs,virtio/gen,bindings.rs,net/gen,benches,logger/,signal_handler.rs,time.rs
+--exclude \
```
Comment (kalyazin) [inline docs/tracing.md:49]: could add indentation for better readability like
```bash
clippy-tracing \
  --action fix \
  --path ./src \
  --exclude \
```

File: docs/tracing.md [hunk 3]
```diff
@@ -33,16 +33,35 @@ outputting a `Trace` level log when entering and exiting every function.
...
+benches,\
+virtio/gen,bindings.rs,net/gen,\
+log-instrument-macros/,log-instrument/,clippy-tracing/,\
+vmm_config/logger.rs,logger/,signal_handler.rs,time.rs
 ```
 
-and re-compile with `--features tracing`:
+`--exclude` can be used to avoid adding instrumentation to specific
+files, here it is used to avoid adding instrumentation in:
```
Comment (kalyazin) [inline docs/tracing.md:57]: looks like 2 separate sentences.

File: docs/tracing.md [hunk 4]
```diff
@@ -33,16 +33,35 @@ outputting a `Trace` level log when entering and exiting every function.
...
+
+- tests.
```
Comment (kalyazin) [inline docs/tracing.md:59]: do we need full stops after every item?

File: src/utils/Cargo.toml [hunk 5]
```diff
@@ -18,10 +18,10 @@ versionize = "0.1.10"
...
 
 [dev-dependencies]
 serde_json = "1.0.99"
 
 [features]
-tracing = ["log-instrument"]
+tracing = []
```
Comment (kalyazin) [inline src/utils/Cargo.toml:27]: commit message `fix: Fix optional dependency issue`: would be helpful to reference log-instrument somehow in the summary.

File: src/utils/Cargo.toml (hunk 1 above)
Comment (JonathanWoollett-Light) [reply src/utils/Cargo.toml:21]: This would be this https://github.com/firecracker-microvm/firecracker/issues/4215 since it is a non-trivial task to write this test.

File: docs/tracing.md (hunk 2 above)
Comment (JonathanWoollett-Light) [reply docs/tracing.md:49]: I personally prefer it without the indents since the additional spaces are not ignored. With 1 space if it's copied and placed into 1 line it looks normal, with 4 space it can look unusual.

File: docs/tracing.md (hunk 3 above)
Comment (JonathanWoollett-Light) [reply docs/tracing.md:57]: To split it into 2 sentences it would need to be reworded
```
`--exclude` can be used to avoid adding instrumentation to specific
files. `--exclude` is used here it is used to avoid adding instrumentation in:
```

File: docs/tracing.md (hunk 4 above)
Comment (JonathanWoollett-Light) [reply docs/tracing.md:59]: I use bullet points as if they could be separate sentences.

So:
```
it is used to avoid adding instrumentation in tests.
it is used to avoid adding instrumentation in bindings.
it is used to avoid adding instrumentation in the instrumentation tooling.
```
becomes:
```
it is used to avoid adding instrumentation in [tests|bindings|the instrumentation tooling].
```
which becomes:
```
it is used to avoid adding instrumentation in:
- tests.
- bindings.
- the instrumentation tooling.
```
This makes the most sense to me, although I do not know a reference for this.

File: docs/tracing.md (hunk 2 above)
Comment (kalyazin) [reply docs/tracing.md:49]: > additional spaces are not ignored

not sure I understand that. Do you mean if we add spaces they are not always rendered?

> with 4 space it can look unusual.

I think 2 spaces would serve both purposes?

File: src/utils/Cargo.toml (hunk 1 above)
Comment (kalyazin) [reply src/utils/Cargo.toml:21]: makes sense. Would be worth adding a comment in the issue?

File: src/utils/Cargo.toml (hunk 5 above)
Comment (JonathanWoollett-Light) [reply src/utils/Cargo.toml:27]: I've updated the commit message to reference it:
```text
When running `cargo build --features tracing` with `src/utils`
instrumented errors are produced due to the `log-instrument` optional
dependency not being enabled by the `utils/tracing` feature. This commit
makes the dependency non-optional to fix this, when unneeded the
dependency will still be ommited at compile time however this may very
slightly increase compile time.
```

File: docs/tracing.md (hunk 3 above)
Comment (kalyazin) [reply docs/tracing.md:57]: > is used here it is used

This is a bit hard to parse. Also, maybe replace "here" with some more explicit reference to the piece above?

Comment (JonathanWoollett-Light) [reply docs/tracing.md:57]: That's my mistake, updated to remove that.

File: docs/tracing.md (hunk 2 above)
Comment (kalyazin) [reply docs/tracing.md]: > This in my experience is not always consistent using the terminal, some terminals will format it as:

It doesn't look like a big concern to me as long as the command is still syntactically valid.

**Review by kalyazin — APPROVED** [review]
